homeassistant/components/pglab/sensor.py

```diff
@@ -0,0 +1,132 @@
+    SensorEntityDescription(
+        key=SENSOR_TEMPERATURE,
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_VOLTAGE,
+        translation_key="voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_REBOOT_TIME,
+        translation_key="runtime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        icon="mdi:progress-clock",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PGLABConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up sensor for device."""
+
+    @callback
+    def async_discover(
+        pglab_device: PyPGLabDevice,
+        pglab_device_sensor: PGLabDeviceSensor,
+    ) -> None:
+        """Discover and add a PG LAB Sensor."""
+        pglab_discovery = config_entry.runtime_data
+        for description in SENSOR_INFO:
+            pglab_sensor = PGLabSensor(
+                pglab_discovery, pglab_device, pglab_device_sensor, description
+            )
+            async_add_entities([pglab_sensor])
+
+    # Register the callback to create the sensor entity when discovered.
+    pglab_discovery = config_entry.runtime_data
+    await pglab_discovery.register_platform(hass, Platform.SENSOR, async_discover)
+
+
+PARALLEL_UPDATES = 0
+
+
+class PGLabSensor(PGLabEntity, SensorEntity):
+    """A PGLab sensor."""
+
+    def __init__(
+        self,
+        pglab_discovery: PGLabDiscovery,
+        pglab_device: PyPGLabDevice,
+        pglab_device_sensor: PGLabDeviceSensor,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the Sensor class."""
+
+        super().__init__(
+            discovery=pglab_discovery,
+            device=pglab_device,
+            entity=pglab_device_sensor.sensors,
+        )
+
+        self._type = description.key
+        self._pglab_device_sensor = pglab_device_sensor
+        self._attr_unique_id = f"{pglab_device.id}_{description.key}"
+        self._state: str | datetime | None = None
+        self.entity_description = description
+
+        if self.entity_description.device_class == SensorDeviceClass.TIMESTAMP:
+            self._state = utcnow()
+        else:
+            self._state = "0"
+
+    @callback
+    def state_updated(self, payload: str) -> None:
+        """Handle state updates."""
+
+        # get the sensor value from pglab multi fields sensor
+        value = self._pglab_device_sensor.state[self._type]
+
+        if self.entity_description.device_class == SensorDeviceClass.TIMESTAMP:
+            self._state = utcnow() - timedelta(seconds=value)
```

Comment:
```suggestion
            self._attr_native_value = utcnow() - timedelta(seconds=value)
```
we can set `self._attr_native_value` directly and remove the `native_value()` method ant the `_state` attribute.

Reply:
I didn't know that.... thank you I'll change the code.